clients/fossid-webapp/src/main/kotlin/Extensions.kt

```diff
@@ -302,3 +310,62 @@ suspend fun FossIdRestService.createIgnoreRule(
             "apply_to" to scope.name.lowercase()
         )
     )
+
+/**
+ * Ask the FossID server to generate a [reportType] report containing [selectionType]. The report will be generated in
+ * the [directory].
+ */
+suspend fun FossIdRestService.generateReport(
+    user: String,
+    apiKey: String,
+    scanCode: String,
+    reportType: ReportType,
+    selectionType: SelectionType,
+    directory: File
+): Result<File> {
+    val response = generateReport(
+        PostRequestBody(
+            "generate_report",
+            SCAN_GROUP,
+            user,
+            apiKey,
+            "scan_code" to scanCode,
+            "report_type" to reportType.name.lowercase(),
+            "selection_type" to selectionType.name.lowercase()
+        )
+    )
+
+    return if (response.isSuccessful) {
+        // The API is quirky here: If the report type is HTML, the result is returned as a standard HTML response,
+        // without filename. For any other report types, the result is returned as an attachment with a filename in the
+        // headers (even for report types generating HTML such as DYNAMIC).
+        val fileName = if (reportType == ReportType.HTML) "${UUID.randomUUID()}-$scanCode-report.html"
```

Comment:
To have a unique report name ?
If you query different reports for the same scan, they would overwrite themselves ?
(but in really they don't because for non-HTML reports, FossID generates a unique identifier in the file name).
So you could see this as a safeguard.

Reply:
I'm not sure what you mean by "different reports". Do you mean different *types* of reports? If so, they should be deduplicated by the file extension.

Or you you saying that even when sticking to a HTML report for the same `scanCode`, the report could vary over time? I'm sorry if my FossID knowledge is lacking here, but I thought a `scanCode` uniquely identified a fixed scan and its result, thus requesting a report for a given `scanCode` should always yield the same result. Is that not the case?

Reply:
Anyway, can we somehow reverse-engineer how the unique ID for the other report formats (which seems to be a SHA1) is calculated by FossID, and do the same for HTML reports? I'd strongly prefer that over a random UUID, because requesting the same report for the same `scanCode` without any changes to the scan should create exactly the same report, including the filename.

Reply:
> I'm not sure what you mean by "different reports". Do you mean different _types_ of reports? If so, they should be deduplicated by the file extension.

Yes I mean report type but unfortunately extension is not enough because both `HTML` and `DYNAMIC` types are html.

Reply:
However I agree with the SHA1 this is a go idea.

Reply:
This would require too much changes to the current logic because we get the file contents only after the fine name has been calculated.
Therefore I removed the unique ID altogether.